pkg/controller/core/clusterqueue_controller.go

```diff
@@ -50,18 +52,31 @@ type ClusterQueueReconciler struct {
 	qManager   *queue.Manager
 	cache      *cache.Cache
 	wlUpdateCh chan event.GenericEvent
+	rfUpdateCh chan event.GenericEvent
 	watchers   []ClusterQueueUpdateWatcher
 }
 
-func NewClusterQueueReconciler(client client.Client, qMgr *queue.Manager, cache *cache.Cache, watchers ...ClusterQueueUpdateWatcher) *ClusterQueueReconciler {
-	return &ClusterQueueReconciler{
+func NewClusterQueueReconciler(
+	client client.Client,
+	qMgr *queue.Manager,
+	cache *cache.Cache,
+	watchers ...ClusterQueueUpdateWatcher,
+) *ClusterQueueReconciler {
+	cqRec := &ClusterQueueReconciler{
 		client:     client,
 		log:        ctrl.Log.WithName("cluster-queue-reconciler"),
 		qManager:   qMgr,
 		cache:      cache,
 		wlUpdateCh: make(chan event.GenericEvent, updateChBuffer),
+		rfUpdateCh: make(chan event.GenericEvent, updateChBuffer),
 		watchers:   watchers,
 	}
+	for _, w := range watchers {
+		if rfRec, match := w.(*ResourceFlavorReconciler); match {
+			rfRec.AddUpdateWatcher(cqRec)
+		}
+	}
```

Comment:
@alculquicondor Does it match your expected implementation?

Reply:
I don't think this loop should be inside the constructor. In theory, not all watchers to ClusterQueue updates need to send updates back to the ClusterQueue reconciler.

I think you can simply do `rfRec.AddUpdateWatcher(cqRec)` in `SetupManagers`

Reply:
Makes sense.

Reply:
Done.